lib/instrumentation/modules/_lambda-handler.js

```diff
@@ -0,0 +1,32 @@
+'use strict'
+const { getLambdaHandlerInfo } = require('../../lambda')
+const Instrumentation = require('../index')
+
+// wraps a nested property on the module object with the lambda handler
+// @param string field ex. "foo.baz.bar"
+// @param object module
+function wrapNestedPath (agent, field, module) {
+  const parts = field.split('.')
+  let object = module
+  while (parts.length > 0) {
+    if (!object) {
+      break
+    }
+    const prop = parts.shift()
+    if (parts.length === 0) {
+      object[prop] = agent.lambda(object[prop])
```

Comment:
Well that's annoying, especially since the version I had previously generally worked. And the edge case we want to fix, `index.nosuchprop`, would result in no lambda function being executed regardless of the agent's presence.  To be honest this doesn't seem worth the time we're taking with it just to preserve an error message in the logs -- that feeling's  probably why I biffed the change.

Regardless -- new version pushed.  The `wrapNestedPath` function has test coverage.  If the function's called with a property that doesn't exist the lambda function won't be wrapped. 

Tested in the live lambda environment and it works.  An `index.notthere` handler results in the following error message in the logs, as expected.

```
2022-03-02T23:43:18.222Z undefined ERROR Uncaught Exception {"errorType":"Runtime.HandlerNotFound","errorMessage":"index.notthere is undefined or not exported","stack":["Runtime.HandlerNotFound: index.notthere is undefined or not exported"," at Object.module.exports.load (/var/runtime/UserFunction.js:246:11)"," at Object.<anonymous> (/var/runtime/index.js:43:30)"," at Module._compile (internal/modules/cjs/loader.js:1085:14)"," at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)"," at Module.load (internal/modules/cjs/loader.js:950:32)"," at Function.Module._load (internal/modules/cjs/loader.js:790:12)"," at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:76:12)"," at internal/main/run_main_module.js:17:47"]} 
```

I'm leaving the lambda wrapper itself alone but wouldn't object if you want to add guard clause there.

If there's more cases you want `wrapNestedPath` to handle they should be pretty easy to add a new test case for.